perllib/FixMyStreet/Roles/Extra.pm

```diff
@@ -243,4 +243,24 @@ sub get_extra_field_value {
     return $field->{value};
 }
 
+=head2 get_extra_field
+
+    my $field = $problem->get_extra_field(name => 'field_name');
+
+Return a field stored in `_fields` in extra, or undefined if it's not present.
+Can use either `name` or `code` to identify the field.
+
+=cut
+
+sub get_extra_field {
+    my ($self, %opts) = @_;
+
+    my @fields = @{ $self->get_extra_fields() };
+
+    my $comparison = $opts{code} ? 'code' : 'name';
```

Comment:
If you add this line to `get_extra_field_value` above, isn't it then the same? :)

Reply:
No, because it returns `$field->{value}`. I wanted the whole `$field`.

Reply:
Hmm, yes, I see; I'd perhaps argue getting rid of get_extra_field_value then (there is a rare bug on reports that fail to send where if it's present but blank it adds more and more entries instead of updating the extra field see e.g. report 1771619), not sure.

Reply:
I dunno, it seems like quite a useful shortcut.